### PR TITLE
Fairscale Integration

### PR DESCRIPTION
**Pull request by amogkam**
Depends on #41

### CODE REVIEW

File: ray_lightning/ray_ddp_sharded.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+if _FAIRSCALE_AVAILABLE:
+    from fairscale.nn.data_parallel.sharded_ddp import ShardedDataParallel
+    from fairscale.optim import OSS
+
+    from pytorch_lightning.overrides.fairscale import \
+        LightningShardedDataParallel, unwrap_lightning_module_sharded
+
+
+class RayShardedPlugin(RayPlugin):
```
Comment (richardliaw) [inline ray_lightning/ray_ddp_sharded.py:19]: would it be possible to do something like:

```
RayPlugin(mode="sharded")
```
?

**Comment by richardliaw** [comment]
Are tests passing properly?

File: ray_lightning/ray_ddp_sharded.py (hunk 1 above)
Comment (amogkam) [reply ray_lightning/ray_ddp_sharded.py:19]: It is possible, though I would prefer to keep it as is. PTL has a separate `DDPPlugin` and `DDPShardedPlugin` so I think using the same dichotomy for these Ray plugins would be more intuitive for users.

**Review by richardliaw — APPROVED** [review]
nice work here!

**Pull request merged**